docs/developer_guide/unit_test_design_guidelines.md

```diff
@@ -120,9 +149,12 @@ def test_sequence_encoders(
 ):
 ```
 
-- Use temporary directories for any generated data.  **Rationale**: Avoids polluting the local file system when testing locally.  **Illustrative code fragment**:
+## Use tempfile for generated data
 
-```
+Use temporary directories for any generated data. PyTest will automatically clean up these directories after test run
```

Comment:
Could be good to mention that `pytest` actually has built-in fixtures for `tmp_path` and `tmpdir`

https://docs.pytest.org/en/6.2.x/tmpdir.html

We use them in some places like `test_experiment.py`, but not everywhere.